contracts/axelar-interchain-token-service/src/contract.rs

```diff
@@ -0,0 +1,100 @@
+use axelar_soroban_std::ensure;
+use soroban_sdk::{contract, contractimpl, Address, Env, String,};
+
+use crate::error::InterchainTokenServiceError;
+use crate::event;
+use crate::storage_types::{DataKey, TrustedItsAddress};
+
+#[contract]
+pub struct InterchainTokenService;
+
+#[contractimpl]
+impl InterchainTokenService {
+    pub fn initialize(
+        env: Env,
+        owner: Address
+    ) -> Result<(), InterchainTokenServiceError> {
+        ensure!(
+            env.storage()
+                .instance()
+                .get(&DataKey::Initialized)
+                .unwrap_or(true),
```

Comment:
```suggestion
                .is_none(),
```
we should ensure it's false/set to none instead, or fail otherwise